tests/compiler/assert-nonnull.untouched.wat

```diff
@@ -306,12 +306,6 @@
   local.get $0
   call_indirect (type $FUNCSIG$i)
   local.tee $1
-  local.tee $2
-  if (result i32)
-   local.get $2
-  else
-   unreachable
-  end
   call $~lib/rt/stub/__retain
```

Comment:
This is strange, as if the result of the indirect call was known to be never null, but it shouldn't be (signature is `() => Foo | null`).